core/lib/common/worker.js

```diff
@@ -233,4 +233,24 @@ module.exports = class Worker {
 		return stdout;
 	}
 
+	async rebootDut (target) {
+		this.logger.log(`Rebooting DUT`);
+		await this.executeCommandInHostOS(
+			`touch /tmp/reboot-check && systemd-run --on-active=2 reboot`,
+			target,
+		);
+		await utils.waitUntil(async () => {
+			return (
+				(await this.executeCommandInHostOS(
+					'[[ ! -f /tmp/reboot-check ]] && echo pass || echo fail',
+					target,
+				)) === 'pass'
+			);
+		}, true);
```

Comment:
having `true` here is risky. It means that if the `executeCommandInHostOS()` fails because the DUT isn't back online yet, an error will be thown (`EHOSTUNREACH`).

Setting it to `false` would mean that this wouldn't happen

Reply:
Oh right, yeah. What's the best reboot instruction we created? I think it's the one we have in Dtoverlay because that being being most recent I think. Will check that one out.

Reply:
Okay, also fixed the condition statement. The updated one is. 

```
await utils.waitUntil(async () => {
			return (
				(await this.executeCommandInHostOS(
					'[[ ! -f /tmp/reboot-check ]] && echo pass',
					target,
				)) === 'pass'
			);
		}, false);
```